[RDY] vim-patch:8.0.0302 NA

ckelsel opened the pull request:
Problem:    Cannot set terminal key codes with :let.
Solution:   Make it work.

https://github.com/vim/vim/commit/e353c402e63b9b0a0bc06acf390e352d9e7eeaeb

ckelsel committed:
fix hostname_spec.lua test failed

ckelsel committed:
test

ckelsel committed:
rm test

ckelsel committed:
Merge remote-tracking branch 'upstream/master'

ckelsel committed:
Merge remote-tracking branch 'upstream/master'

DarkDefender committed:
tui: fix DECSCUSR logic #6997

Fix linuxvt cursor shape codes
Fix konsole cursor_shapes (even when inside tmux)
Do not trust old VTE terminal lies

Closes #6978
Closes #7002
Closes #7049

ckelsel committed:
ignore patch-2367,2364

ckelsel committed:
merge upstream/master

ckelsel committed:
merge upstream

ckelsel committed:
Merge remote-tracking branch 'upstream/master'

ckelsel committed:
Merge remote-tracking branch 'upstream/master'

justinmk committed:
vim-patch:8.0.0518

Closes #7086

Problem:    Storing a zero byte from a multi-byte character causes fold text
            to show up wrong.
Solution:   Avoid putting zero in ScreenLines. (Christian Brabandt,
            closes vim/vim#1567)

https://github.com/vim/vim/commit/c6cd8409c2993b1476e123fba11cb4b8d743b896

justinmk committed:
vim-patch:8.0.0311

Problem:    Linebreak tests are old style.
Solution:   Turn the tests into new style. Share utility functions. (Ozaki
            Kiichi, closes vim/vim#1444)

https://github.com/vim/vim/commit/544d3bc9f0e494cb712a33b61558b8e8e12b1e0b

justinmk committed:
vim-patch:8.0.0090 fix breakindent bug

(original Vim commit-message is bogus)

https://github.com/vim/vim/commit/6c896867c4f5d759616028ef7cbfce2a9ed32600

justinmk committed:
vim-patch:8.0.0290 vim-patch:8.0.0394

vim-patch:8.0.0290: cursor positioning wrong if wide character wraps
Problem:    If a wide character doesn't fit at the end of the screen line, and
            the line doesn't fit on the screen, then the cursor position may
            be wrong. (anliting)
Solution:   Don't skip over wide character. (Christian Brabandt, closes vim/1408)

vim-patch:8.0.0394
Problem:    Tabs are not aligned when scrolling horizontally and a Tab doesn't
            fit. (Axel Bender)
Solution:   Handle a Tab as a not fitting character. (Christian Brabandt)
            Also fix that ":redraw" does not scroll horizontally to show the
            cursor.  And fix the test that depended on the old behavior.

https://github.com/vim/vim/commit/abc39ab642791ae3d22a524516eeedb673a95d9d

justinmk committed:
vim-patch:8.0.0126

Problem:    Display problem with 'foldcolumn' and a wide character.
            (esiegerman)
Solution:   Don't use "extra" but an allocated buffer. (Christian Brabandt,
            closes vim/vim#1310)

https://github.com/vim/vim/commit/6270660611a151c5d0f614a5f0248ccdc80ed971

justinmk committed:
vim-patch:8.0.0128

Problem:    Display test fails on MS-Windows.
Solution:   Set 'isprint' to "@".

https://github.com/vim/vim/commit/7089237885218eb8a19805bc2b75481c4efcd6ba

justinmk committed:
vim-patch:8.0.0524

Problem:    Folds are messed up when 'encodin' is "utf-8".
Solution:   Also set the fold character when it's not multi-byte.

https://github.com/vim/vim/commit/8da1e6cedf839902e15987a98733ebd31b5f1b81

ckelsel committed:
Merge remote-tracking branch 'upstream/master'

ckelsel committed:
Merge remote-tracking branch 'upstream/master'

ckelsel committed:
Merge remote-tracking branch 'upstream/master'

ckelsel committed:
Merge remote-tracking branch 'upstream/master'

ckelsel committed:
Merge remote-tracking branch 'upstream/master'

ckelsel committed:
Merge remote-tracking branch 'upstream/master'

ckelsel committed:
Merge remote-tracking branch 'upstream/master'

ckelsel committed:
Merge remote-tracking branch 'upstream/master'

justinmk committed:
oldtest: cannot `:set term` in Nvim

ckelsel committed:
Merge remote-tracking branch 'upstream/master'

justinmk committed:
vim-patch:8.0.0235

Problem:    Memory leak detected when running tests for diff mode.
Solution:   Free p_extra_free.

https://github.com/vim/vim/commit/b031c4ea04eb1e37a873fbb85e90d835aa1e2b1c

jamessan committed:
travis: Move TSAN to last stage and allow failure

TSAN build has been much less reliable lately, so it shouldn't hold up
the other tests.

ZyX-I committed:
lua/executor: Fix crash when printing empty string (#7157)

justinmk committed:
io: more guards against NULL filename (#7159)

References ac055d677aa9
References #4370

ckelsel committed:
Merge remote-tracking branch 'upstream/master' into HEAD

ckelsel committed:
Merge remote-tracking branch 'upstream/master'

ckelsel committed:
Merge remote-tracking branch 'upstream/master'

ckelsel committed:
Merge remote-tracking branch 'upstream/master'

ckelsel committed:
Merge remote-tracking branch 'upstream/master'

ckelsel committed:
Merge remote-tracking branch 'upstream/master'

ckelsel committed:
Merge remote-tracking branch 'upstream/master'

ckelsel committed:
Merge remote-tracking branch 'upstream/master'

ckelsel committed:
vim-patch:8.0.0302 NA

Neovim removed the ability to set termcodes directly.

Problem:    Cannot set terminal key codes with :let.
Solution:   Make it work.

https://github.com/vim/vim/commit/e353c402e63b9b0a0bc06acf390e352d9e7eeaeb

ckelsel commented:
```find_termcode, add_termcode, ttest```
I cann't find correct functions to replace above functions.

jamessan commented:
That's because Neovim removed the ability to set termcodes directly.

```
	*t_xx* *:set-termcap* *termcap-options* *t_AB* *t_Sb* *t_vb* *t_SI*
Nvim does not have special `t_XX` options nor <t_XX> keycodes to configure
terminal capabilities. Instead Nvim treats the terminal as any other UI.
```

This patch should be NA.

ckelsel committed:
vim-patch:8.0.0302 NA

Neovim removed the ability to set termcodes directly.

Problem:    Cannot set terminal key codes with :let.
Solution:   Make it work.

https://github.com/vim/vim/commit/e353c402e63b9b0a0bc06acf390e352d9e7eeaeb
